packages/back-end/src/services/experimentResults.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  computeRiskValues,
+  ExperimentMetricInterface,
+} from "shared/experiments";
+import { ExperimentSnapshotInterface } from "back-end/types/experiment-snapshot";
+import { ExperimentInterface } from "back-end/types/experiment";
+import { DifferenceType } from "back-end/types/stats";
+import { findDimensionById } from "back-end/src/models/DimensionModel";
+
+type ExperimentResultRow = {
+  experimentName: string;
+  experimentId: string;
+  snapshotId: string;
+  snapshotDate: string | null;
+  dimensionId: string | null;
+  dimensionName: string | null;
+  dimensionValue: string | null;
+  metricName: string | null;
+  metricId: string;
+  baselineVariationName: string | null;
+  baselineVariationId: string | null;
+  baselineVariationUsers: number | null;
+  baselineVariationNumerator: number | null;
+  baselineVariationDenominator: number | null;
+  baselineVariationMean: number | null;
+  baselineVariationStdDev: number | null;
+  variationName: string | null;
+  variationId: string | null;
+  variationUsers: number;
+  variationNumerator: number;
+  variationDenominator: number | null;
+  variationMean: number;
+  variationStdDev: number | null;
+  differenceType: DifferenceType;
+  effect: number | null;
+  chanceToWin: number | null;
+  relativeRisk: number | null;
+  pValue: number | null;
+  pValueAdjusted: number | null;
+  ciLower: number | null;
+  ciUpper: number | null;
+  ciLowerAdjusted: number | null;
+  ciUpperAdjusted: number | null;
+};
+
+export async function getExperimentResultRows({
+  experiment,
+  snapshot,
+  metricMap,
+  dimension,
+}: {
+  experiment: ExperimentInterface;
+  snapshot: ExperimentSnapshotInterface;
+  metricMap: Map<string, ExperimentMetricInterface>;
+  dimension?: string;
+}): Promise<ExperimentResultRow[]> {
+  const dimensionName = dimension
+    ? (await findDimensionById(dimension, experiment.organization))?.name ||
+      dimension?.split(":")?.[1] ||
+      dimension
+    : null;
+
+  const rows: ExperimentResultRow[] = [];
+
+  snapshot.analyses.forEach((analysis) => {
+    // skip analyses where the control variation is not the baseline variation
+    if (analysis.settings.baselineVariationIndex !== 0) {
+      return;
+    }
```

Comment:
Would CUPED cause duplicate rows?  I forget if we store CUPED on/off as 2 separate analyses or not.  If they are separate analyses, I think that would result in duplicate copies of every row.  Are there any other analysis settings like that we should watch out for?

Reply:
>I forget if we store CUPED on/off as 2 separate analyses or not

We don't right now. I want to hook that up at some point, but it's a bit complicated as the CUPED logic is squirrely and if you toggle it off in the UI it actually turns off the setting in the analysis form.

There are no other duplicates because right now the only thing that varies across analyses are `differenceType` and `baselineVariationIndex`.